### PR TITLE
International elements

### PR DESCRIPTION
**Pull request by praveenrach**
### JIRA link ###
https://tools.hmcts.net/jira/browse/PRL-2302

### Change description ###
Citizen Journey
Save International Elements link fields present under Respond to Application Link in Respondent Task List page.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/tasklistresponse/miam/MIAMMapper.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { CaseWithId } from '../../../app/case/case';
+import { Miam, Respondent, YesOrNo } from '../../../app/case/definition';
+import type { AppRequest } from '../../../app/controller/AppRequest';
+
+export const setMIAMDetails = (respondent: Respondent, req: AppRequest): Respondent => {
+  let miamFromResponsent: Miam;
+  if (respondent?.value?.response && respondent?.value?.response?.miam) {
```
Comment (cshiwarale) [inline src/main/steps/tasklistresponse/miam/MIAMMapper.ts:7]: Do we need both condition checks? or only one 'respondent?.value?.response?.miam' can suffice?

File: src/main/steps/tasklistresponse/start/utils.ts [hunk 2]
```diff
@@ -31,17 +31,28 @@ export const getConsentToApplicationStatus = (userCase: Partial<CaseWithId> | un
...
+
+  return status;
 };
 
-export const getInternationalFactorsStatus = (userCase: Partial<CaseWithId> | undefined): SectionStatus => {
+export const getInternationalFactorsStatus = (
+  userCase: Partial<CaseWithId> | undefined,
+  userIdamId: string
+): SectionStatus => {
+  console.log('userIdamId===>' + userIdamId);
```
Comment (cshiwarale) [inline src/main/steps/tasklistresponse/start/utils.ts:55]: please avoid print sensitive user info in logs

File: sonar-project.properties [hunk 3]
```diff
@@ -6,7 +6,7 @@ sonar.tests=src/test/
...
...tail.ts,src/main/app/document/DeleteDocumentRequest.ts,src/main/app/document/DocumentManagementClient.ts,src/main/app/document/GenerateAndUploadDocumentRequest.ts,src/main/steps/applicant/task-list/tasklist.ts,src/main/steps/applicant/yourdocuments/alldocuments/alldocuments/tasklistalldocuments.ts,src/main/app/document/DocumentManagementController.ts,src/main/steps/applicant/applicantCaseSequence.ts,src/main/steps/respondent/task-list/tasklist.ts,src/main/steps/respondent/yourdocuments/alldocuments/alldocuments/tasklistalldocuments.ts,src/**/*.test.ts,**/__mocks__/**,src/**/*content.ts,src/**/components/*.ts,src/**/definition.ts,src/**/summary/util.ts,src/**/summary/util.ts,src/**/steps/constants.ts,src/**/willingness-to-attend-miam/miam-cost-exemptions.ts,src/**/steps/index.ts,src/**/steps/urls.ts,src/**/case/CaseApi.ts,src/**/app/controller/*.ts,src/**/checkanswers/ConfirmContactDetailsGetController.ts,src/**/respondent/task-list/utils.ts,src/**/respondent/respondentcaseSequence.ts,src/main/steps/common/summary/utils.ts,src/**/app/controller/GetCaseController.ts,src/**/app/controller/GetRespondentCaseController.ts,src/main/steps/prl-cases/dashboard/controller/GetCaseController.ts
```
[the remark this answers is not in the document]
Comment (praveenrach) [reply sonar-project.properties:9]: This Controller is not part of this jira, so excluded from Sonar Checks

File: src/main/steps/respondent/task-list/tasklist.ts [hunk 4]
```diff
@@ -12,7 +12,8 @@ import {
 } from './utils';
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types */
 
-export const generateRespondentTaskList = (sectionTitles, taskListItems, userCase) => {
+export const generateRespondentTaskList = (sectionTitles, taskListItems, userCase, userIdamId) => {
+  console.log(userIdamId);
```
[the remark this answers is not in the document]
Comment (praveenrach) [reply src/main/steps/respondent/task-list/tasklist.ts:16]: will remove the loggers

File: src/main/steps/respondent/respondentcaseSequence.test.ts [hunk 5]
```diff
@@ -2,75 +2,75 @@ import { respondentCaseSequence } from './respondentcaseSequence';
...
-
-    expect(respondentCaseSequence[15].url).toBe('/respondent/task-list');
-    expect(respondentCaseSequence[15].showInSection).toBe('aboutRespondentCase');
-    expect(respondentCaseSequence[15].getNextStep({})).toBe('/respondent/yourdocuments/alldocuments/orders');
-
-    expect(respondentCaseSequence[16].url).toBe('/respondent/yourdocuments/alldocuments/orders');
-    expect(respondentCaseSequence[16].showInSection).toBe('aboutRespondentCase');
-    expect(respondentCaseSequence[16].getNextStep({})).toBe('/respondent/task-list');
+    expect(respondentCaseSequence).toHaveLength(83);
+    // expect(respondentCaseSequence[0].url).toBe('/respondent/task-list');
```
[the remark this answers is not in the document]
Comment (praveenrach) [reply src/main/steps/respondent/respondentcaseSequence.test.ts:6]: will remove the commented code

File: src/main/steps/tasklistresponse/international-factors/InternationalFactorsMapper.ts [hunk 6]
```diff
@@ -0,0 +1,116 @@
+import { CaseWithId } from '../../../app/case/case';
+import { CitizenInternationalElements, Respondent, YesOrNo } from '../../../app/case/definition';
+import type { AppRequest } from '../../../app/controller/AppRequest';
+
+export const setInternationalFactorsDetails = (respondent: Respondent, req: AppRequest): Respondent => {
+  let internationaElementsFromRespondent: CitizenInternationalElements;
```
[the remark this answers is not in the document]
Comment (praveenrach) [reply src/main/steps/tasklistresponse/international-factors/InternationalFactorsMapper.ts:6]: corrected to internationalElementsFromRespondent

File: src/main/steps/tasklistresponse/miam/MIAMMapper.ts (hunk 1 above)
Comment (praveenrach) [reply src/main/steps/tasklistresponse/miam/MIAMMapper.ts:7]: will remove this

File: src/main/steps/tasklistresponse/international-factors/InternationalFactorsMapper.ts [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+  if (respondent?.value?.response && respondent?.value?.response?.citizenInternationalElements) {
```
[the remark this answers is not in the document]
Comment (praveenrach) [reply src/main/steps/tasklistresponse/international-factors/InternationalFactorsMapper.ts:8]: will update the code

**Review by cshiwarale — APPROVED** [review]

**Pull request merged**